api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java

```diff
@@ -209,27 +222,74 @@ public void handleSubmission(FormEntrySession formEntrySession, HttpServletReque
+
+            FormEntryContext formEntrycontext = formEntrySession.getContext();
+
+            Set<org.openmrs.Diagnosis> existingDiagnoses = new HashSet<org.openmrs.Diagnosis>(Context.getDiagnosisService().getDiagnoses(formEntrycontext.getExistingPatient(), null));
+            Set<org.openmrs.Diagnosis> resubmittedDiagnoses = new HashSet<org.openmrs.Diagnosis>();
+
             for (JsonNode node : list) {
                 CodedOrFreeTextAnswer answer = new CodedOrFreeTextAnswer(node.get("diagnosis").getTextValue(), conceptService);
                 Diagnosis.Order diagnosisOrder = Diagnosis.Order.valueOf(node.get("order").getTextValue());
                 Diagnosis.Certainty certainty = Diagnosis.Certainty.valueOf(node.get("certainty").getTextValue());
+                Integer existingDiagnosis = null;
 
                 if (node.path("existingObs").getNumberValue() != null) {
                     JsonNode nd = node.get("existingObs");
                     System.out.println(nd);
                 }
 
-                org.openmrs.Diagnosis diagnosis = new org.openmrs.Diagnosis();
-                diagnosis.setEncounter(formEntrySession.getEncounter());
-                diagnosis.setCertainty(certainty == Diagnosis.Certainty.CONFIRMED ? ConditionVerificationStatus.CONFIRMED : ConditionVerificationStatus.PROVISIONAL);
-                diagnosis.setRank(diagnosisOrder == Diagnosis.Order.PRIMARY ? 1 : 2);
-                diagnosis.setPatient(formEntrySession.getPatient());
-                diagnosis.setDiagnosis(new CodedOrFreeText(answer.getCodedAnswer(), answer.getSpecificCodedAnswer(), answer.getNonCodedAnswer()));
+                if (node.path("existingDiagnosis").getNumberValue() != null) {
+                    existingDiagnosis = node.get("existingDiagnosis").getIntValue();
+                }
+
+                org.openmrs.Diagnosis diagnosis;
+
+                Integer orderId = diagnosisOrder == Diagnosis.Order.PRIMARY ? 1 : 2;
+                ConditionVerificationStatus certaintyStatus = certainty == Diagnosis.Certainty.CONFIRMED ? ConditionVerificationStatus.CONFIRMED : ConditionVerificationStatus.PROVISIONAL;
 
-                diagnosis.getEncounter().setEncounterType(Context.getEncounterService().getEncounterTypeByUuid("d7151f82-c1f3-4152-a605-2f9ea7414a79"));
-                Context.getEncounterService().saveEncounter(diagnosis.getEncounter());
-                Context.getDiagnosisService().save(diagnosis);
+                if(existingDiagnosis !=null){
+
+                    diagnosis = Context.getDiagnosisService().getDiagnosis(existingDiagnosis);
+
+                    // Check if an existing diagnosis is being resubmitted
+                    if (!existingDiagnoses.add(diagnosis)) {
+                        resubmittedDiagnoses.add(diagnosis);
+                    }
+
+                    if(!diagnosis.getRank().equals(orderId) || !diagnosis.getCertainty().equals(certaintyStatus)){
```

Comment:
Is it only certainty and rank that the UI lets you change?

Reply:
@dkayiwa Those are all the attributes I can see. am I missing something here?